src/fastcs/launch.py

```diff
@@ -100,12 +104,43 @@ def run(self):
 
     async def serve(self) -> None:
         coros = [self._backend.serve()]
-        coros.extend([transport.serve() for transport in self._transports])
+        if self._transports:
+            context = {"controller": self._backend}
+            for transport in self._transports:
+                context.update(transport.context())
```

Comment:
We should have a checked update that errors if a key already exists. Hopefully it will never come up, but if it does we need to know.